File: server/src/main/java/com/smartexpense/api/domain/repository/ExpenseRepository.java
package com.smartexpense.api.domain.repository;

import com.smartexpense.api.domain.model.Expense;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface ExpenseRepository {
    List<Expense> findAllByUserIdOrderByExpenseDateDesc(UUID userId);
    Optional<Expense> findByIdAndUserId(UUID id, UUID userId);
    Expense save(Expense expense);
    void deleteByIdAndUserId(UUID id, UUID userId);
}

File: server/src/main/kotlin/com/smartexpense/api/domain/repository/ExpenseRepository.kt
package com.smartexpense.api.domain.repository

import com.smartexpense.api.domain.model.Expense
import java.util.UUID

interface ExpenseRepository {
    fun findAllByUserIdOrderByExpenseDateDesc(userId: UUID): List<Expense>
    fun findByIdAndUserId(id: UUID, userId: UUID): Expense?
    fun save(expense: Expense): Expense
    fun deleteByIdAndUserId(id: UUID, userId: UUID)
}
